kv: improve a test

andreimatei opened the pull request:
This test was mixing a new tracer with a server that was using its own
tracer. Mixing tracers like this in a single trace will not be supported
any more.

Release note: None

andreimatei committed:
kv: improve a test

This test was mixing a new tracer with a server that was using its own
tracer. Mixing tracers like this in a single trace will not be supported
any more.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72940)
<!-- Reviewable:end -->

andreimatei committed:
kv: don't mix tracers in a couple of tests

These tests were using a new tracer to trace operations that were going
into a Server. The Server has its own tracer that it uses. Combining
spans created by two tracers into a single trace like this is fairly
broken (spans can leak because it's not clear what registry they belong
to), and will be prohibited.

Release note: None

andreimatei reviewed (commented):
I added another small commit, if you don't mind taking a look.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72940#-:-Moosvop6dVQ_4MTytqY:bce0k17)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72940#-:-MooxlUaDcC5gmq-lsO8:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72940#-:-MosZ6P161738MyOe_mh:b-ydi9ay)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72940#-:-MoyyU86-Tj_Gi-DUme3:b-ydi9ay)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72940#-:-MozR5DH8YNoHsoNkqc2:b-ydi9ay)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->
